kbs_protocol/src/lib.rs

```diff
@@ -136,19 +136,16 @@ impl KbsProtocolWrapper {
 #[async_trait]
 impl KbsRequest for KbsProtocolWrapper {
     async fn http_get(&mut self, url_string: String) -> Result<Vec<u8>> {
-        let url = Url::parse(&url_string)
+        let _ = Url::parse(&url_string)
             .map_err(|e| anyhow!("Invalid Request URL {url_string}: {e}"))?;
 
-        // Use default port of KBS: 8080
-        let port = url.port().unwrap_or(8080);
-        let host_str = url.host_str().ok_or_else(|| anyhow!("No KBS host given"))?;
-        let kbs_url = format!("{}://{host_str}:{port}", url.scheme());
+        let root_url = url_string.split(KBS_URL_PREFIX).collect::<Vec<&str>>()[0].to_string();
```

Comment:
Is it possible that the provided `url_string` does not have a substr `KBS_URL_PREFIX`? Is the returned vector as expected?